dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePlugin.kt

```diff
@@ -184,17 +184,20 @@ class DdAndroidGradlePlugin : Plugin<Project> {
 
     internal fun resolveExtensionConfiguration(
         extension: DdExtension,
-        flavorName: String
+        variant: ApplicationVariant
     ): DdExtensionConfiguration {
-        val flavorConfig = extension.variants.findByName(flavorName)
-
-        return DdExtensionConfiguration().apply {
-            versionName = flavorConfig?.versionName ?: extension.versionName
-            serviceName = flavorConfig?.serviceName ?: extension.serviceName
-            site = flavorConfig?.site ?: extension.site
-            checkProjectDependencies = flavorConfig?.checkProjectDependencies
-                ?: extension.checkProjectDependencies
+        val configuration = DdExtensionConfiguration()
+        configuration.updateWith(extension)
+
+        val flavors = variant.productFlavors.map { it.name }
+        val iterator = VariantIterator(flavors)
+        iterator.forEach {
+            val config = extension.variants.findByName(it)
```

Comment:
this won't match the case of `buildFreeDebug` (which is mentioned in the related task), because `debug` is not a part of flavor, but of build type.